src/components/Tooltip/Tooltip.js

```diff
@@ -163,13 +163,26 @@ export default class Tooltip extends Component {
      */
     triggerText: PropTypes.node,
 
+    /**
+     * The callback function to optionally render the icon element.
+     * It should be a component with React.forwardRef().
+     */
+    renderIcon: function(props, propName, componentName) {
+      if (props[propName] == undefined) return;
+
+      if (props[propName].$$typeof !== Symbol.for('react.forward_ref'))
```

Comment:
Not sure if the assumption of the React internal is safe for future React. Let's see what others think.

Reply:
I have tried `react-is` with `isForwardRef` but it didn't work well; the other solution could be to accept any React element but the developer will never know that a ref is necessary.

Reply:
just curious, what kind of issues was `isForwardRef` causing? I'm also a bit iffy on using React internal API here so I guess we'll just have to keep an eye on this over time

Reply:
Hi @emyarod, thanks for the question, `react-is` use the same approach used here, though while debugging I found that they consider first that the given data should be an `Element` and then a `forwardRef`, maybe I found a bug, reason because of doing ` ReactIs.isForwardRef(props[propName])` always returns `false`. 

I'll open an issue on `react` to confirm this behavior.

https://github.com/facebook/react/blob/master/packages/react-is/src/ReactIs.js#L30

![image](https://user-images.githubusercontent.com/1231888/52968470-44d03300-337b-11e9-90c5-42affb391ad9.png)

Reply:
@emyarod Added a new commit, the trick was to create the element with `const RefForwardingComponent = props[propName]` then `<RefForwardingComponent/>`; this as a parameter for `isForwardRef` will pass the evaluation.

Reply:
awesome thanks for the clarification and glad you figured it out @devniel! and I guess we won't have to file an issue over on the React repo either